Dockerfile

```
@@ -0,0 +1,3 @@
+FROM golang:1.18-alpine
+COPY tagger /usr/bin/tagger
+ENTRYPOINT ["/usr/bin/tagger", "--config", "/etc/tagger/tagger.yml"]
```

```suggestion
ENTRYPOINT ["/usr/bin/tagger"]
```

> An ENTRYPOINT allows you to configure a container that will run as an executable.

https://docs.docker.com/engine/reference/builder/#entrypoint

> The main purpose of a CMD is to provide defaults for an executing container. These defaults can include an executable, or they can omit the executable, in which case you must specify an ENTRYPOINT instruction as well.
> If CMD is used to provide default arguments for the ENTRYPOINT instruction, both the CMD and ENTRYPOINT instructions should be specified with the JSON array format.

https://docs.docker.com/engine/reference/builder/#cmd

Since we're intending to use the container as a wrapper around the executable, I think we should be keeping the `ENTRYPOINT` to just the binary itself. If we want to provide default flags, we can do that via the `CMD` arrary

---

Using prometheus as an example, here's their Dockerfile:
https://github.com/prometheus/prometheus/blob/main/Dockerfile#L24-L28

---

Using prometheus as an example, here is their Dockerfile:
https://github.com/prometheus/prometheus/blob/main/Dockerfile#L24-L28

---

TIL
>You can use the exec form of ENTRYPOINT to set fairly stable default commands and arguments and then use either form of CMD to set additional defaults that are more likely to be changed.